src/services/NumberService.ts

```diff
@@ -0,0 +1,235 @@
+        let cvsIdNumber = testNumber.substring(1, 3);
+        let certLetter = testNumber.substring(3, 4);
+        let sequenceNumber = testNumber.substring(4, 7);
+
+
+        if (parseInt(sequenceNumber, 10) === 999) {
+            sequenceNumber = "001";
+            if (certLetter === "Z") {
+                certLetter = "A";
+                if (parseInt(cvsIdNumber, 10) === 99) {
+                    cvsIdNumber = "01";
+                    cvsIdLetter = String.fromCharCode(cvsIdLetter.charCodeAt(0) + 1);
+                } else {
+                    cvsIdNumber = (parseInt(cvsIdNumber, 10) + 1).toString().padStart(2, "0");
+                }
+            } else {
+                certLetter = String.fromCharCode(certLetter.charCodeAt(0) + 1);
+            }
+
+        } else {
+            sequenceNumber = (parseInt(sequenceNumber, 10) + 1).toString().padStart(3, "0");
+        }
+
+        let newTestNumber = cvsIdLetter + cvsIdNumber + certLetter + sequenceNumber;
+        newTestNumber = this.appendCheckSumToTestNumber(newTestNumber);
+
+        const newTestNumberObject: TestNumber = {
+            id: cvsIdLetter + cvsIdNumber,
+            certLetter,
+            sequenceNumber,
+            testNumber: newTestNumber,
+            testNumberKey: 1
+        };
+
+        return newTestNumberObject;
+    }
+
+    /**
+     * Calculates and creates the next trailerId object based on the last trailerId
+     * @param trailerIdObject - last trailerId
+     */
+    public createNextTrailerIdObject(trailerIdObject: TrailerId): TrailerId {
+        const newSequenceNumber: number = trailerIdObject.sequenceNumber + 1;
+        const newTrailerId = trailerIdObject.trailerLetter + newSequenceNumber;
+        const newTrailerIdObject: TrailerId = {
+            trailerId: newTrailerId,
+            trailerLetter: trailerIdObject.trailerLetter,
+            sequenceNumber: newSequenceNumber,
+            testNumberKey: 2
+        };
+        return newTrailerIdObject;
+    }
+
+    /**
+     * Appends calculated checksum to the test number
+     * @param testNumber - the test number
+     */
+    public appendCheckSumToTestNumber(testNumber: string) {
+        const testNumberWithoutLetters = testNumber.replace(/\D/g, "");
+
+        const firstLetterAlphabeticalIndex = (testNumber.charCodeAt(0) - 64);
+        const secondLetterAlphabeticalIndex = (testNumber.charCodeAt(3) - 64);
+
+        const checkSum = firstLetterAlphabeticalIndex
+            + parseInt(testNumberWithoutLetters.charAt(0), 10)
+            + parseInt(testNumberWithoutLetters.charAt(1), 10) * 3
+            + secondLetterAlphabeticalIndex
+            + parseInt(testNumberWithoutLetters.charAt(2), 10)
+            + parseInt(testNumberWithoutLetters.charAt(3), 10) * 3
+            + parseInt(testNumberWithoutLetters.charAt(4), 10);
+
+        let stringCheckSum;
+        if (checkSum < 10) {
+            stringCheckSum = checkSum.toString().padStart(2, "0");
```

Comment:
Might need to compare this to Develop's TestNumberService. There was a padding bug which Alex  solved a few days ago. Specifically, the `.padStart(2, "0")` from 225 needs to go at the end in order to apply to 227 as well (for  cases where "checksum - 100" < 10)

Reply:
It says my branch is up to date with develop when I try to do a rebase, I don't think that bug was merged into develop yet.

Reply:
fixed